### PR TITLE
Upgrade React to v18

### LINKED ISSUE #2187: React 18 support?
When can we expect React 18 support (mainly around updating peer dependencies and adopting new hydration methods?)

### PR DESCRIPTION
**Pull request by becca-bailey**
Upgrades React to v18, and updates peer dependencies to work with React 18. (Are there any reasons for us to remove support for v16 right now?)

This also updates a few misc. dependency issues that came up when running `yarn audit`. 

Fixes #2187

### CODE REVIEW

**Comment by becca-bailey** [comment]
@scottrippey I didn't update any lint rules in this PR, but for some reason `eslint-typescript` is throwing a bunch of warnings in CI. Any ideas why?

File: package.json [hunk 1]
```diff
@@ -107,13 +107,13 @@
...
-    "react-dom": "^16.13.1",
+    "react": "^18.1.0",
+    "react-dom": "^18.1.0",
     "react-hot-loader": "v2.0.0-alpha-4",
-    "react-test-renderer": "^16.13.1",
+    "react-test-renderer": "^18.1.0",
     "rimraf": "^3.0.2",
     "seedrandom": "^2.4.3",
-    "storybook-addon-performance": "^0.10.0",
+    "styled-components": "^5.3.5",
```
Comment (scottrippey) [inline package.json:116]: Why was this added?

File: packages/victory-area/package.json [hunk 2]
```diff
@@ -25,7 +25,7 @@
     "victory-vendor": "^36.4.0"
   },
   "peerDependencies": {
-    "react": "^16.6.0 || ^17.0.0"
+    "react": "^16.6.0 || ^17.0.0 || ^18.0.0"
```
Comment (scottrippey) [inline packages/victory-area/package.json:28]: Maybe we should change this to;
```suggestion
    "react": ">=16.8.0"
```
For people who are early-adopters of React, this is helpful.

File: package.json (hunk 1 above)
Comment (becca-bailey) [reply package.json:116]: We are using `styled-components` in some of our storybook files. I'm not sure why the imports were all working before!

File: packages/victory-area/package.json (hunk 2 above)
Comment (becca-bailey) [reply packages/victory-area/package.json:28]: 👍

**Review by scottrippey — APPROVED** [review]
Looking good!

**Comment by stur86** [comment]
Great! When can we expect a new release on NPM including this fix?

**Comment by stur86** [comment]
Thanks, that's fantastic!
